### ISSUE TITLE
FEATURE REQUEST: week view, events to always stack from top down

### ISSUE THREAD

**Issue by gazman99** [mattlewis92/angular-bootstrap-calendar#99]
Is or would it be feasible to have to the events in week view to always stack from top down irrespective to what the start hour is. Obviously if a event spans multiple days, the top position is adjusted to the next vertical slot.

**Comment by mattlewis92** [comment]
Sorry I don't quite get what you mean

**Comment by mattlewis92** [comment]
Oh wait I get what you mean now, I don't think it's a particularly trivial thing to add but I'll take a look.

**Comment by gazman99** [comment]
Hey matt, so what i'm meaning is: currently every event has a new row, starting from day 1 in the week view through to day 7 in week view. e.g If day 1 has 3 events, it will use 3 rows. So day 2 will start on row 4. 
Can it start again on row 1.....is that any clearer?

**Comment by mattlewis92** [comment]
I looked into this as it's really difficult to add I'm afraid so I'm not going to add it. Sorry I couldn't be of more help.

**Comment by gazman99** [comment]
Thanks for the feedback. I've been asked to implement this myself. So maybe I'll send u a pull request if I manage a solution. Probably go down the route of a different template and css. I will keep you posted.

**Comment by cyberdemon8** [comment]
Also I'm interested in this enhancement... :smiley:

**Comment by cvcv01** [comment]
Not sure what gazman99 is looking for exactly, but it would be nice to have the weekly view look like the daily view, except,it would show 7 columns (days) instead of just 1 column (day). That way you can see the entire week (by the hour) in one glimpse.

**Comment by eduardomanski** [comment]
Hi!

I've been looking at <a href="https://github.com/Serhioromano/bootstrap-calendar">Twitter Bootstrap Calendar</a> (where this project came from), and now they have the week view like day view, with hours. I don't know about the alingment of events in different days in week view.

I'll try to write here, hope to achieve that and then try to help you.

**Comment by cyberdemon8** [comment]
@cvcv01 if you set this flag to true in a config block you obtain what you want
```javascript
angular.module('myModule')
  .config(function(calendarConfigProvider) {
    calendarConfigProvider.showTimesOnWeekView(true);
  });
```
Only two problems:
- on fully overlapping events only the last event is visible
- every event is one "row" height (the end datetime is not considered)

**Comment by cvcv01** [comment]
Oh ok, I see it now, and I get the problems you mentionned.

I've added 3 events on the same day: AAA Event, BBB Event and CCC. AAA Event and BBB Event are set to the same time.

Here's how it looks in monthly view:
![cal1](https://cloud.githubusercontent.com/assets/13452790/8972871/51026698-362c-11e5-8e11-4a7d24cba196.jpg)


Here's the weekly view:
![cal2](https://cloud.githubusercontent.com/assets/13452790/8972881/6993f17c-362c-11e5-9a7e-bc945a2ec7a5.jpg)


On the weekly view, AAA and BBB Events are stacked over each other (instead of side by side like on the daily view), plus they are only one "row" height.

On the daily view, the display is good.
![cal3](https://cloud.githubusercontent.com/assets/13452790/8972946/2377f8b8-362d-11e5-9b17-6920463232ed.jpg)

Each column (day) from the weekly view should act like a single column of the daily view. But then again, the width of the events in the daily view are static and not responsive.

Any way of making it responsive?

**Comment by cvcv01** [comment]
Not sure if you guy have seen this calendar. http://angular-ui.github.io/ui-calendar/  Not the best UI, but you can stack a lot of events side by side on the daily and weekly view. Solid.

**Comment by cvcv01** [comment]
@eduardomanski Cool, I guess you're talking about this. https://github.com/Serhioromano/bootstrap-calendar/issues/401 It looks nice, I agree.

**Comment by mattlewis92** [comment]
If you don't mind the jquery dependency then I would say use the ui-calendar for this kind of functionality - it's actually a really difficult problem to get all events to stack when overlapping and then you have the additional problem of when events span multiple days they need to appear in multiple columns. Hence why this feature is implemented behind a flag and has the issues that @cyberdemon8 highlighted above.

**Comment by cvcv01** [comment]
I understand, its a complex task.

For now, even if the daily view is not responsive, it displays nicely. All the events are stacked properly.

For the weekly view (with hours), in the current code, I think it's too complex to position the events under each day. For the weekly view (with hours), instead of having one parent div element (position:relative) for the entire week, I think we should have one parent div element (set to position:relative) per day, then each event is postionned (postion:absolute) within that parent day div.

**Comment by freefri** [comment]
+1 to this enhancement

**Comment by immaraj** [comment]
How to show each day event  in each coulmn. Now it  shows like 2nd  day event start s only after the completion of 1 day event
